tools/monorepo-utils/src/code-freeze/commands/changelog/lib/index.ts

```diff
@@ -54,8 +54,8 @@ const updateReleaseChangelogs = async (
 
 	// Convert PR number to markdown link.
 	nextLog = nextLog.replace(
-		/\[#(\d+)\]/g,
-		'[$&](https://github.com/woocommerce/woocommerce/pull/$1)'
+		/\[#(\d+)\](?!\()/g,
+		'[#$1](https://github.com/woocommerce/woocommerce/pull/$1)'
```

Comment:
Since we're only matching a PR number anyway let's just go ahead and be strict about the output.